src/sentry/tasks/check_auth.py

```diff
@@ -12,52 +16,83 @@
 from sentry.silo.base import SiloMode
 from sentry.tasks.base import instrumented_task
 from sentry.utils import metrics
+from sentry.utils.env import in_test_environment
 
 logger = logging.getLogger("sentry.auth")
 
 AUTH_CHECK_INTERVAL = 3600 * 24
 
 
+def find_providers_requiring_refresh() -> List[str]:
+    from sentry.auth import manager
+
+    return [name for name, provider in manager if provider.requires_refresh]
+
+
 @instrumented_task(name="sentry.tasks.check_auth", queue="auth.control", silo_mode=SiloMode.CONTROL)
-def check_auth(**kwargs):
+def check_auth(chunk_size=100, **kwargs):
     """
-    Iterates over all accounts which have not been verified in the required
-    interval and creates a new job to verify them.
+    Checks for batches of auth identities and schedules them to refresh in a batched job.
+    That batched job can recursively trigger check_auth to continue processing auth identities if necessary.
+    Updates last_synced as it schedules batches so that further calls generally select non overlapping batches.
     """
     # TODO(dcramer): we should remove identities if they've been inactivate
     # for a reasonable interval
     now = timezone.now()
-    chunk_size = 100
     cutoff = now - timedelta(seconds=AUTH_CHECK_INTERVAL)
     identity_ids_list = list(
         AuthIdentity.objects.using_replica()
         .filter(last_synced__lte=cutoff)
-        .values_list("id", flat=True)
+        .values_list("id", flat=True)[:chunk_size]
     )
-    for n in range(0, len(identity_ids_list), chunk_size):
-        identity_ids_chunk = identity_ids_list[n : n + chunk_size]
-        with unguarded_write(router.db_for_write(AuthIdentity)):
-            AuthIdentity.objects.filter(id__in=identity_ids_chunk).update(last_synced=now)
 
-        for identity_id in identity_ids_chunk:
-            check_auth_identity.apply_async(
-                kwargs={"auth_identity_id": identity_id}, expires=AUTH_CHECK_INTERVAL
-            )
+    if identity_ids_list:
+        with unguarded_write(router.db_for_write(AuthIdentity)):
+            AuthIdentity.objects.filter(id__in=identity_ids_list).update(last_synced=now)
+        check_auth_identity.apply_async(
+            kwargs={"auth_identity_ids": identity_ids_list, "chunk_size": chunk_size},
+            expires=AUTH_CHECK_INTERVAL,
+        )
```

Comment:
this is not roll forward compatible -- the current task definition takes one positional argument